<launch>

  <arg name="model" default="$(find ur_description)/urdf/ur.urdf.xacro name:=ur5" />

  <arg name="root_link" default="shoulder_link" />
  <arg name="tip_link" default="wrist_3_link" />

  <arg name="chain_joints_file" default="$(find manipulability_metrics_examples)/config/ur5/ur5_joints.yaml" />
  <arg name="zeros" default="$(find manipulability_metrics_examples)/config/ur5/ur5_zeros.yaml" />

  <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />

  <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui">
    <rosparam param="zeros" file="$(arg zeros)" />
  </node>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <!-- Open rviz with predefined settings -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find manipulability_metrics_examples)/config/ur5/ur5_demo.rviz" required="true" />

  <!-- manipulability_metrics -->
  <node name="ur5_manipulability_demo" pkg="manipulability_metrics_examples" type="ur5_manipulability_demo" output="screen">
    <param name="root_link" value="$(arg root_link)" />
    <param name="tip_link" value="$(arg tip_link)" />
    <rosparam param="chain_joints" file="$(arg chain_joints_file)" />
  </node>

</launch>
